### PR TITLE
Cleanup deprecated  Director API

### PR DESCRIPTION
**Pull request by kjaksik**
**Description**

Changes proposed in this pull request:

- Switch to packages api
- Switch to correct mutation and query names in asa api
- Remove apiRuntimeAuth type

**Related issue(s)**
#1040

### CODE REVIEW

File: components/director/internal/domain/package/mock/api.go [hunk 1]
```diff
@@ -121,19 +121,17 @@ components:
           type: string
 `)
 	return &graphql.APIDefinition{
```
Comment (kfurgol) [inline components/director/internal/domain/package/mock/api.go:123]: I believe whole `components/director/internal/domain/package/mock` directory can be deleted

File: components/director/internal/model/document.go [hunk 2]
```diff
@@ -5,14 +5,13 @@ import (
...
 type Document struct {
-	ApplicationID *string
-	PackageID     *string
-	ID            string
-	Tenant        string
-	Title         string
-	DisplayName   string
-	Description   string
-	Format        DocumentFormat
+	PackageID   *string
```
Comment (pkosiec) [inline components/director/internal/model/document.go:8]: string, not *string

File: components/director/internal/model/api.go [hunk 3]
```diff
@@ -7,21 +7,16 @@ import (
...
-	ID            string
-	ApplicationID *string
-	PackageID     *string
-	Tenant        string
-	Name          string
-	Description   *string
-	Spec          *APISpec
-	TargetURL     string
+	ID          string
+	PackageID   *string
```
Comment (pkosiec) [inline components/director/internal/model/api.go:11]:
```suggestion
	PackageID   string
```

File: components/director/pkg/graphql/document.go [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
 type Document struct {
-	ID            string         `json:"id"`
-	ApplicationID *string        `json:"applicationID"`
-	PackageID     *string        `json:"packageID"`
-	Title         string         `json:"title"`
-	DisplayName   string         `json:"displayName"`
-	Description   string         `json:"description"`
-	Format        DocumentFormat `json:"format"`
+	ID          string         `json:"id"`
+	PackageID   *string        `json:"packageID"`
```
Comment (pkosiec) [inline components/director/pkg/graphql/document.go:5]: PackageID should be string

File: components/director/pkg/graphql/event_api.go [hunk 5]
```diff
@@ -1,11 +1,10 @@
 package graphql
 
 type EventDefinition struct {
-	ID            string  `json:"id"`
-	ApplicationID *string `json:"applicationID"`
-	PackageID     *string `json:"packageID"`
-	Name          string  `json:"name"`
-	Description   *string `json:"description"`
+	ID          string  `json:"id"`
+	PackageID   *string `json:"packageID"`
```
Comment (pkosiec) [inline components/director/pkg/graphql/event_api.go:5]: PackageID should be string

File: components/director/internal/domain/scenarioassignment/resolver.go [hunk 6]
```diff
@@ -28,8 +28,7 @@ type Service interface {
 	Delete(ctx context.Context, in model.AutomaticScenarioAssignment) error
 }
 
-// TODO: Change order of params: Service before Converter
-func NewResolver(transact persistence.Transactioner, converter Converter, svc Service) *Resolver {
+func NewResolver(transact persistence.Transactioner, svc Service, converter Converter) *Resolver {
```
Comment (pkosiec) [inline components/director/internal/domain/scenarioassignment/resolver.go:31]: 👌

File: components/director/internal/domain/eventdef/entity.go [hunk 7]
```diff
@@ -10,7 +10,6 @@ type Entity struct {
 	ID          string         `db:"id"`
 	TenantID    string         `db:"tenant_id"`
 	PkgID       sql.NullString `db:"package_id"`
```
Comment (pkosiec) [inline components/director/internal/domain/eventdef/entity.go:12]: please change it to string

File: components/director/internal/domain/document/entity.go [hunk 8]
```diff
@@ -7,7 +7,6 @@ import (
 type Entity struct {
 	ID          string         `db:"id"`
 	TenantID    string         `db:"tenant_id"`
-	AppID       sql.NullString `db:"app_id"`
 	PkgID       sql.NullString `db:"package_id"`
```
Comment (pkosiec) [inline components/director/internal/domain/document/entity.go:10]: PackageID should be string

File: components/director/internal/domain/api/entity.go [hunk 9]
```diff
@@ -9,13 +9,11 @@ import (
 type Entity struct {
 	ID          string         `db:"id"`
 	TenantID    string         `db:"tenant_id"`
-	AppID       sql.NullString `db:"app_id"`
 	PkgID       sql.NullString `db:"package_id"`
```
Comment (pkosiec) [inline components/director/internal/domain/api/entity.go:12]: PackageID should be string

File: components/director/pkg/graphql/schema.graphql [hunk 10]
```diff
@@ -952,7 +913,7 @@ type Viewer {
 
 type Webhook {
 	id: ID!
-	applicationID: ID! @deprecated(reason: "Use ID field from parent object")
+	applicationID: ID!
```
Comment (pkosiec) [inline components/director/pkg/graphql/schema.graphql:916]: 👌  good catch

File: components/director/pkg/graphql/schema.graphql [hunk 11]
```diff
@@ -952,7 +913,7 @@ type Viewer {
 
 type Webhook {
```
Comment (pkosiec) [inline components/director/pkg/graphql/schema.graphql:914]: Please also remove `UNUSED` enum value from `FetchRequestStatus`. Thanks!

File: components/director/pkg/graphql/schema.graphql [hunk 12]
```diff
@@ -1161,41 +1109,18 @@ type Mutation {
...
 	- [add event definition to package](examples/add-event-definition-to-package/add-event-definition-to-package.graphql)
 	"""
 	addEventDefinitionToPackage(packageID: ID!, in: EventDefinitionInput! @validate): EventDefinition! @hasScopes(path: "graphql.mutation.addEventDefinitionToPackage")
-	"""
-	**Examples**
-	- [update event definition](examples/update-event-definition/update-event-definition.graphql)
-	"""
 	updateEventDefinition(id: ID!, in: EventDefinitionInput! @validate): EventDefinition! @hasScopes(path: "graphql.mutation.updateEventDefinition")
-	"""
-	**Examples**
```
Comment (pkosiec) [inline components/director/pkg/graphql/schema.graphql:1182]: Why? Please restore these valid examples. Thanks

Comment (pkosiec) [reply components/director/pkg/graphql/schema.graphql:1182]: The only examples that should be remove are these which belong to removed GraphQL API.

**Pull request merged**